### PR TITLE
webpack 4 support

### PR DESCRIPTION
**Pull request by Bnaya**
On webpack 4, the minimum node version is 6 (lts/boron)

### CODE REVIEW

**Comment by Bnaya** [comment]
i've published it here:
https://www.npmjs.com/package/@brandtotal/webpack-s3-plugin

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,9 @@
 {
-  "presets": ["es2015"]
-}
-
+  "presets": [
+    ["env", {
+      "targets": {
+        "node": "6.10"
```
Comment (Bnaya) [inline .babelrc]: On webpack 4, the minimum node version is 6 (lts/boron)

**Comment by Bnaya** [comment]
The tests are passing locally. the CI is missing the env params

**Comment by Bnaya** [comment]
@MikaAK friendly ping :)

File: src/s3_plugin.js [hunk 2]
```diff
@@ -84,29 +88,27 @@ module.exports = class S3Plugin {
                              compiler.options.output.context ||
                              '.'
 
-    compiler.plugin('after-emit', (compilation, cb) => {
+    compiler.hooks.afterEmit.tapPromise(packageJson.name, async(compilation) => {
```
Comment (MikaAK) [inline src/s3_plugin.js]: Marking these functions async is unnecessary no?

Comment (Bnaya) [reply src/s3_plugin.js]: It makes sure that if they will throw, it will turn into Promise.reject

**Comment by MikaAK** [comment]
Need to get this merged and then we can begin merging in prs, struggling with appveyor, i know how to get travis to pass but if anyone has insight on appveyor help would be appreciated!
https://github.com/webpack-contrib/s3-plugin-webpack/pull/96

**Comment by Bnaya** [comment]
As i can see, the problem is that, because it's my PR and i'm not in the team, the secrets are not being used.
https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml

**Comment by Bnaya** [comment]
try to push my code as a branch on this repo, and open new PR with it

**Comment by MikaAK** [comment]
For Travis yes that's the problem however I can solve that one easily. I am having a hard time with appveyor however, and that's the real problem. I reached out to the webpack team for support but never heard anything back

**Comment by Bnaya** [comment]
I'm not using appveyor, so i have no idea

**Comment by Sunshine168** [comment]
emm , So it still break in webpack4 now?

**Comment by Bnaya** [comment]
@Sunshine168 you may use https://www.npmjs.com/package/@brandtotal/webpack-s3-plugin

**Comment by Sunshine168** [comment]
@Bnaya  thx, i am already switch to it

**Comment by Bnaya** [comment]
@MikaAK has the appveyor issues solved? i can rebase my PR if you will merge it

**Comment by MikaAK** [comment]
Yep feel free to rebase!

**Comment by Bnaya** [comment]
I can do that, but i wonder why :)

**Comment by MikaAK** [comment]
I'm going to rewrite it to rollup, however since I'm doing so, I would rather change the build right now and just do it then

**Comment by Bnaya** [comment]
i'de say its redundant, but i will do so:)

**Comment by Bnaya** [comment]
superseded by #101

**Comment by Bnaya** [comment]
i've deprecated our fork package, please use the original package

**Pull request closed without merging**